juniper/src/vespa/juniper/sumdesc.cpp

```diff
@@ -66,10 +129,20 @@ int complete_word(unsigned char* start, ssize_t length,
     // Figure out if a word needs completion or if we are just going
     // to eliminate whitespace
     if (!wordchar(ptr)) {
-        whitespace_elim = true;
-        // Change direction of scan
-        increment = -increment;
-        chartest = wordchar;
+        if (increment > 0 && il_ann_anchor_char(ptr)) {
```

Comment:
It would be nice with an extension of the description at L129-130 to match the extra checks added.